sw/device/lib/dif/dif_base.h

```diff
@@ -53,6 +53,16 @@ typedef enum dif_result {
    * Indicates that the Ip's FIFO (if it has one or more of) is full.
    */
   kDifIpFifoFull = 5,
+  /**
+   * Indicates that the attempted operation would attempt a read/write to an
+   * address that would go out of range.
+   */
+  kDifOutOfRange = 6,
+  /**
+   * Indicates that the attempted operation would attempt a read/write to an
+   * address that is not aligned.
+   */
+  kDifUnaligned = 7,
```

Comment:
@sriyerg @moidx Just want to point your attention to these additions to deal with some of the specific errors reported in memory interfacing IPs like the otp_ctrl. I tried to follow the conventions [here](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/lib/absl_status.h). Wasn't sure if also adding `kDifUnaligned` was worth it or not. Thoughts?

Reply:
In what context will this be used? An out of bounds address will result in an exception (IMO the default exception handler should treat that as an error and abort immediately to aid quicker debug). Unaligned access may or may not, so the outcomes for both are possibly different.

Reply:
see below

Reply:
D'oh! The large diffs in `dif_otp_ctrl.c|h` were not rendered, so my previous search for `kDifUnaligned` did not return anything. 

Seems ok to me, given that the diagnosis for both is different - being able to differentiate between the two seems helpful. Though, I left a question in the comment below.

Reply:
SGTM